apps/arena/lib/arena/configuration.ex

```diff
@@ -65,4 +66,12 @@ defmodule Arena.Configuration do
       skill
     end
   end
+
+  defp parse_mechanic_fields({:leap, attrs}) do
+    {:leap, %{attrs | on_arrival_mechanic: parse_mechanic_config(attrs.on_arrival_mechanic)}}
```

Comment:
Why are we parsing just the leap on arrival mechanics? maybe you can make this method more generic doing:
```
defp parse_mechanic_fields({skill, attrs}) do
{skill, %{attrs | on_arrival_mechanic: parse_mechanic_config(attrs.on_arrival_mechanic)}}
end
```